packages/react/src/components/Modal/Modal.base.tsx

```diff
@@ -411,6 +411,10 @@ export const ModalBase: React.FunctionComponent<IModalProps> = React.forwardRef<
 
     useUnmount(() => {
       internalState.events.dispose();
+      if (internalState.onModalCloseTimer) {
+        clearTimeout(internalState.onModalCloseTimer);
+        internalState.onModalCloseTimer = 0;
```

Comment:
For consistency, should we also be clearing this state after line 387? (As its optionally set afterwards)

Should we also clear registerInitialModalPosition timeout here in case the component unmounts as prop is set to open? (Unlikely edge case)

Reply:
Possibly. There are also and raf and timeout on L391 that should probably be cleared as well.

However, since we don't have a repro (just a partner's error telemetry) I'd prefer to make minimal changes to see if we can resolve the issue. This case seems the most likely source of the problem to me as it's a relatively long-lived timer.

WDYT?